src/profile-logic/profile-data.js

```diff
@@ -1507,6 +1507,47 @@ export function isNetworkMarker(marker: TracingMarker): boolean {
   return !!(marker.data && marker.data.type === 'Network');
 }
 
+export function getNetworkMarkers(thread: Thread): Array<NetworkPayload> {
+  const { markers } = thread;
+  const networkMarker = [];
+
+  // making sure we return all markers after we got all from type network
+  if (!markers) {
+    return networkMarker;
+  }
+
+  // filter for all markers
+  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+    const data = markers.data[markerIndex];
+    if (!data || data.type !== 'Network') {
+      continue;
+    }
+
+    // Match start marker with stop / redirect, if they have one.
+    if (
+      (!!networkMarker && data.status === 'STATUS_STOP') ||
```

Comment:
I believe `!!networkMarker` will always be true. `networkMarker` will always be an array.